src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -0,0 +1,150 @@
+    {
+        private IDisposable _optionsReloadToken;
+
+        public JsonConsoleFormatter(IOptionsMonitor<JsonConsoleFormatterOptions> options)
+            : base (ConsoleFormatterNames.Json)
+        {
+            ReloadLoggerOptions(options.CurrentValue);
+            _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
+        }
+
+        public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider scopeProvider, TextWriter textWriter)
+        {
+            string message = logEntry.Formatter(logEntry.State, logEntry.Exception);
+            if (logEntry.Exception == null && message == null)
+            {
+                return;
+            }
+            LogLevel logLevel = logEntry.LogLevel;
+            string category = logEntry.Category;
+            int eventId = logEntry.EventId.Id;
+            Exception exception = logEntry.Exception;
+            const int DefaultBufferSize = 1024;
+            var output = new ArrayBufferWriter<byte>(DefaultBufferSize);
+            using (var writer = new Utf8JsonWriter(output, FormatterOptions.JsonWriterOptions))
+            {
+                writer.WriteStartObject();
+                string timestamp = null;
+                var timestampFormat = FormatterOptions.TimestampFormat;
+                if (timestampFormat != null)
+                {
+                    var dateTime = FormatterOptions.UseUtcTimestamp ? DateTime.UtcNow : DateTime.Now;
+                    timestamp = dateTime.ToString(timestampFormat);
+                }
+                writer.WriteString("Timestamp", timestamp);
+                writer.WriteNumber(nameof(logEntry.EventId), eventId);
+                writer.WriteString(nameof(logEntry.LogLevel), GetLogLevelString(logLevel));
+                writer.WriteString(nameof(logEntry.Category), category);
+                writer.WriteString("Message", message);
+
+                if (exception != null)
+                {
+                    writer.WriteStartObject(nameof(Exception));
+                    writer.WriteString(nameof(exception.Message), exception.Message.ToString());
+                    writer.WriteString("Type", exception.GetType().ToString());
+                    writer.WriteStartArray(nameof(exception.StackTrace));
+                    string stackTrace = exception?.StackTrace;
+                    if (stackTrace != null)
+                    {
+                        foreach (var stackTraceLines in stackTrace?.Split(Environment.NewLine))
+                        {
+                            writer.WriteStringValue(stackTraceLines);
+                        }
+                    }
+                    writer.WriteEndArray();
+                    writer.WriteNumber(nameof(exception.HResult), exception.HResult);
+                    writer.WriteEndObject();
+                }
+
+                if (logEntry.State is IReadOnlyCollection<KeyValuePair<string, object>> stateDictionary)
```

Comment:
Here and below, we're assuming `IReadOnlyCollection<KeyValuePair<string, object>>` is a dictionary, which would typically have unique keys and work for JSON properties.  It is not, an IReadOnlyCollection can have duplicates.  Do we care?

Reply:
Maybe add a test to try to abuse the logger this way and see what it does.